### ISSUE TITLE
`deepinv.Trainer.test` modularisation

### ISSUE THREAD

**Issue by Andrewwango** [deepinv/deepinv#253]
The test method of the Trainer class should use the same methods as the train method, so that when methods (e.g. get samples) in the Trainer are overriden, the test functionality is modified too. 

`deepinv.test` should be deprecated in the same way as `deepinv.train`. It should simply call `deepinv.Trainer.test`, but retain the same functionality as before for compatibility.

**Comment by Andrewwango** [comment]
Eg desired outcome:

```python
class CustomTrainer(Trainer):
    def get_samples_offline(...)
        ...

trainer = CustomTrainer(...)
trainer.train()
...

trainer.test(test_dataloader)
#I want this to use my new overloaded method

```

**Comment by Andrewwango** [comment]
@tachella is this solved by #280 ?

**Comment by tachella** [comment]
Yes indeed
